.github/workflows/nightly-pipeline.yaml

```
@@ -118,7 +118,7 @@ jobs:
           trigger_workflow: true
           wait_workflow: true
   raft-tests:
-    needs: [get-run-info, raft-build]
+    needs: [get-run-info, raft-build, ucx-py-tests]
```

```suggestion
    needs: [get-run-info, raft-build, ucx-py-build]
```
Generally I don't think we should have tests depending on each other.

The issue with depending on `ucx-py-tests` is that `ucx-py-tests` depend on `ucx-py-build`.

So if the `ucx-py-build` fails, then `ucx-py-tests` never runs and therefore `raft-tests` will also never run.

---

I'm not really sure how I feel about saying raft tests passed when using a broken ucx-py package (broken in the sense that it built but its tests didn't pass), but I guess that's no different than our running Python builds and tests against C++ packages before cpp tests ran so at least it's consistent.

---

Yeah. There are some optimizations that need to be done to this pipeline.

Currently each of these pipeline jobs trigger a downstream job, but don't propagate the result of that downstream job to this top-level pipeline.

This is fine for now. We'll optimize in the future.